prereise/gather/griddata/hifld/data_process/transmission.py

```diff
@@ -0,0 +1,171 @@
+from powersimdata.utility.distance import haversine
+
+from prereise.gather.griddata.hifld.data_access import load
+
+
+def check_for_location_conflicts(substations):
+    """Check for multiple substations with identical lat/lon.
+
+    :param pandas.DataFrame substations: data frame of substations.
+    :raises ValueError: if multiple substations with identical lat/lon.
+    """
+    num_substations = len(substations)
+    num_lat_lon_groups = len(substations.groupby(["LATITUDE", "LONGITUDE"]))
+    if num_lat_lon_groups != num_substations:
+        num_collisions = num_substations - num_lat_lon_groups
+        raise ValueError(
+            f"There are {num_collisions} substations with duplicate lat/lon values"
+        )
+
+
+def get_substations_with_nonzero_lines(substations):
+    """Filter substations with LINES attribute equal to zero, and report the number
+    dropped.
+
+    :param pandas.DataFrame substations: data frame of all substations.
+    :return: (*pandas.DataFrame*) -- substations with non-zero values for LINES.
+    """
+    num_substations = len(substations)
+    num_substations_without_lines = len(substations.query("LINES == 0"))
+    print(
+        f"dropping {num_substations_without_lines} substations "
+        f"of {num_substations} total due to LINES parameter equal to 0"
+    )
+
+    return substations.query("LINES != 0").copy()
+
+
+def filter_lines_with_unavailable_substations(lines):
+    """Filter lines with SUB_1 or SUB_2 attribute equal to 'NOT AVAILABLE', and report
+    the number dropped.
+
+    :param pandas.DataFrame lines: data frame of all lines.
+    :return: (*pandas.DataFrame*) -- lines with available substations.
+    """
+    num_lines = len(lines)
+    filtered = lines.query("SUB_1 == 'NOT AVAILABLE' or SUB_2 == 'NOT AVAILABLE'")
+    num_filtered = len(filtered)
+    print(
+        f"dropping {num_filtered} lines with one or more substations listed as "
+        f"'NOT AVAILABLE' out of a starting total of {num_lines}"
+    )
+    return lines.query("SUB_1 != 'NOT AVAILABLE' and SUB_2 != 'NOT AVAILABLE'").copy()
```

Comment:
Same comment as above, `copy` is redundant here. Also, we might wanna keep the same expression of filtering logic to make it straightforward to read, i.e. either `A or B` or `not A and not B`.